src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs

```diff
@@ -56,9 +56,36 @@ internal class DefaultStorageCredentialsValidator : IStorageCredentialsValidator
             catch
             {
                 string message = String.Format(CultureInfo.CurrentCulture,
-                    "The account credentials for '{0}' are incorrect.", account.Credentials.AccountName);
+                    "Invalid storage account '{0}'. Please make sure your credentials are correct.",
+                    account.Credentials.AccountName);
                 throw new InvalidOperationException(message);
             }
+
+            if (isPrimaryAccount)
+            {
+                // Primary storage accounts require Queues
+                var queueClient = account.CreateQueueClient();
+                var queue = queueClient.GetQueueReference("name");
+                try
+                {
+                    await queue.ExistsAsync(cancellationToken);
+                    _primaryCredentials = account.Credentials;
+                }
+                catch (OperationCanceledException)
+                {
+                    throw;
+                }
+                catch (StorageException exception)
+                {
+                    if (!exception.IsNotFoundQueueNotFound())
+                    {
+                        string message = String.Format(CultureInfo.CurrentCulture,
+                            "Invalid storage account '{0}'. The primary storage account (AzureWebJobsStorage) must be a general "
```

Comment:
If it turns out this also applies to Dashboard accounts, then we can generalize the wording to:  "... Primary storage accounts must be general purpose accounts, not restricted blob storage accounts."